spec/prelude/main.md

```diff
@@ -53,17 +53,35 @@ macro_rules! throw_memory_leak {
         do yeet TerminationInfo::MemoryLeak
     };
 }
+
 macro_rules! throw_ill_formed {
-    () => {
-        do yeet TerminationInfo::IllFormed
+    ($($tt:tt)*) => {
+        do yeet TerminationInfo::IllFormed(format!(
+            "{} at {}:{}:{}",
+            format!($($tt)*),
+            file!(),
+            line!(),
+            column!()
+        ))
     };
 }
+
 macro_rules! throw_deadlock {
     () => {
         do yeet TerminationInfo::Deadlock
     };
 }
 
+macro_rules! ensure {
```

Comment:
The macro name is a bit too general to live globally. Maybe call it `ensure_wf`?

Reply:
With the changes to `throw_ill_formed` this does not have to be a global macro anymore.
I made a function out of it (`ensure_wf`) and moved it to `well-formed.md`.